src/plugins/explore/public/application/legacy/discover/application/components/visualizations/utils/use_visualization_types.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { LineChartStyleControls } from '../line/line_vis_config';
+import { IFieldType, IndexPattern } from '../../../../opensearch_dashboards_services';
+import { OpenSearchSearchHit } from '../../../doc_views/doc_views_types';
+import { LineVisStyleControlsProps } from '../line/line_vis_options';
+import { OPENSEARCH_FIELD_TYPES, OSD_FIELD_TYPES } from '../../../../../../../../../data/common';
+import { IExpressionLoaderParams } from '../../../../../../../../../expressions/public';
+import { ExploreVisColumn } from '../types';
+import { visualizationRegistry } from '../visualization_registry';
+
+export interface VisualizationType {
+  readonly name: string;
+  readonly type: string;
+  readonly ui: {
+    style: {
+      defaults: LineChartStyleControls;
+      render: ({
+        styleOptions,
+        onStyleChange,
+        numericalColumns,
+        categoricalColumns,
+        dateColumns,
+      }: LineVisStyleControlsProps) => JSX.Element;
+    };
+  };
+  readonly toExpression: (
+    searchContext: IExpressionLoaderParams['searchContext'],
+    indexPattern: IndexPattern,
+    transformedData?: Array<Record<string, any>>,
+    numericalColumns?: ExploreVisColumn[],
+    categoricalColumns?: ExploreVisColumn[],
+    dateColumns?: ExploreVisColumn[],
+    styleOptions?: Partial<LineChartStyleControls>
+  ) => Promise<string | undefined>;
+}
+
+export type ExploreVisFieldType = 'numerical' | 'categorical' | 'date' | 'unknown';
+
+// Map both OSD_FIELD_TYPES and OPENSEARCH_FIELD_TYPES to ExploreVisFieldType
+// We also need to handle the case where a new field is created with a opensearch type
+const FIELD_TYPE_MAP: Partial<Record<string, ExploreVisFieldType>> = {
+  // Map OSD_FIELD_TYPES to ExploreVisFieldType
+  [OSD_FIELD_TYPES.BOOLEAN]: 'categorical',
```

Comment:
we should prolly add a new optional field to the osdfieldtypes that has this info. and set it with the osd field type factory set these values

Reply:
yes we should move this mapping out of the explore plugin and just build the visFieldType as part of `osdFieldType`. Refactoring this change involves making changes to multiple functions in data plugin such as use_search(), getDataFrames(), are you okay with making this change as part of a follow-up PR? @kavilla